### PR TITLE
Limited availability products

### PR DESCRIPTION
**Pull request by DelusionalLogic**
Now products can have a limited number. Hopefully we can get this in before flan

Also includes some nice order abstraction to hopefully ease multibuy development!

### CODE REVIEW

File: stregsystem/admin.py [hunk 1]
```diff
@@ -112,10 +112,19 @@ def get_price_display(self, obj):
...
+        if (
+            obj.active
+            and (
+                obj.deactivate_date is None
+                or obj.deactivate_date > timezone.now()
+            ) and (
+                obj.quantity is None
+                or obj.quantity > obj.bought
+            )
+        ):
```
Comment (fishbacon) [inline stregsystem/admin.py]: If possible change this if to a return and have the different conditions spelled out in variables rather than this.

Comment (DelusionalLogic) [reply stregsystem/admin.py]: You're right. Product should obviously have this behavior as a member method. I'll get that done!

Comment (fishbacon) [reply stregsystem/admin.py]: Since you are going to change it anyway, what about a descriptive name for instead of `obj`? (eg. product)

Comment (DelusionalLogic) [reply stregsystem/admin.py]: Good point, done!

I also took the liberty of refactoring the expression out into some intermediate variables. It's much more clear what the business logic is now.

**Comment by DelusionalLogic** [comment]
> Looks like this also effectively introduces most of multibuy? Okay.

It does, so it's pretty key to get this right. I'm happy you had the time to look it though. Thanks :)

**Pull request merged**